snow/networking/handler/handler.go

```diff
@@ -435,23 +437,30 @@ func (h *handler) handleSyncMsg(ctx context.Context, msg Message) error {
 	)
 	h.resourceTracker.StartProcessing(nodeID, startTime)
 	h.ctx.Lock.Lock()
+	lockAcquiredTime := h.clock.Time()
 	defer func() {
 		h.ctx.Lock.Unlock()
 
 		var (
-			endTime        = h.clock.Time()
-			histogram      = h.metrics.messages[op]
-			processingTime = endTime.Sub(startTime)
+			endTime           = h.clock.Time()
+			messageHistograms = h.metrics.messages[op]
+			acquireLockTime   = lockAcquiredTime.Sub(startTime)
+			processingTime    = endTime.Sub(lockAcquiredTime)
+			handlingTime      = endTime.Sub(startTime)
 		)
 		h.resourceTracker.StopProcessing(nodeID, endTime)
-		histogram.Observe(float64(processingTime))
+		messageHistograms.handlingTime.Observe(float64(handlingTime))
+		messageHistograms.acquireLockTime.Observe(float64(acquireLockTime))
+		messageHistograms.processingTime.Observe(float64(processingTime))
```

Comment:
Open to naming changes - splitting this into handling time and processing time made sense to me

Reply:
Should we leave processingTime as is (eg for continuity in metrics) and then add acquireLockTime and handlingTime (where processingTime = handlingTime + acquireLockTime)?

Reply:
Yup, that's what is happening here. I just renamed the variables in the code, since `handlingTime` seemed like a better name for the total time spent handling the message vs. actually processing it once the lock was acquired.

This means that `handlingTime = processingTime + acquireLockTime`, but the original metric, which we called `processingTime` still represents the same value.